routes/notes.js

```diff
@@ -1,50 +1,103 @@
-'use strict';
+'use strict'
 
-const express = require('express');
+const express = require('express')
+const mongoose = require('mongoose')
 
-const router = express.Router();
+const router = express.Router()
 
-/* ========== GET/READ ALL ITEMS ========== */
-router.get('/', (req, res, next) => {
+const {MONGODB_URI} = require('../config')
 
-  console.log('Get All Notes');
-  res.json([
-    { id: 1, title: 'Temp 1' },
-    { id: 2, title: 'Temp 2' },
-    { id: 3, title: 'Temp 3' }
-  ]);
+const Note = require('../models/note')
 
-});
+/* ========== GET/READ ALL ITEMS ========== */
+router.get('/', (req, res, next) => {
+  const {searchTerm} = req.query
+  const {folderId} = req.body
+  let filter = {}
+  if (folderId) {
+    filter.folderId = {$regex: folderId, $options: 'i'}
+  }
+  if (searchTerm) {
+    filter.title = {$regex: searchTerm, $options: 'i'}
+  }
+  return Note.find(filter)
+    .sort({updatedAt: 'desc'})
+    .then(results => {
+      res.json(results)
+    })
+    .catch(err => {
+      console.error(err)
+      console.error(`ERROR: ${err.message}`)
+      res.sendStatus(500)
+    })
+})
 
 /* ========== GET/READ A SINGLE ITEM ========== */
 router.get('/:id', (req, res, next) => {
-
-  console.log('Get a Note');
-  res.json({ id: 1, title: 'Temp 1' });
-
-});
+  const id = req.params.id
+
+  if (mongoose.Types.ObjectId.isValid(id)) {
+    return Note.findById(id)
+      .then(results => {
+        res.json(results)
```

Comment:
* Populate tags and folders (optional)
* And handle id not found by calling next()